libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -301,34 +293,35 @@ - (void) syncDown:(SFSyncState*)sync updateSync:(SyncUpdateBlock)updateSync fail
     __block NSUInteger totalSize = 0;
     __block NSUInteger progress = 0;
     __block SFSyncDownTargetFetchCompleteBlock continueFetchBlockRecurse = ^(NSArray *records) {};
-    __weak typeof(self) weakSelf = self;
-    
+
+    __block NSOrderedSet* idsToSkip = nil;
+    if (mergeMode == SFSyncStateMergeModeLeaveIfChanged) {
+        idsToSkip = [target getIdsToSkip:self soupName:soupName];
+    }
+
     SFSyncDownTargetFetchCompleteBlock startFetchBlock = ^(NSArray* records) {
         totalSize = target.totalSize;
         updateSync(nil, totalSize == 0 ? 100 : 0, totalSize, kSyncManagerUnchanged);
         if (totalSize != 0) continueFetchBlockRecurse(records);
     };
 
     SFSyncDownTargetFetchCompleteBlock continueFetchBlock = ^(NSArray* records) {
-        __strong typeof(weakSelf) strongSelf = weakSelf;
         if (records != nil) {
+            // Figure out records to save
+            NSArray* recordsToSave = idsToSkip && idsToSkip.count > 0 ? [self removeWithIds:records idsToSkip:idsToSkip idField:target.idFieldName] : records;
+
+            // Save to smartstore.
+            [target saveRecordsToLocalStore:self soupName:soupName records:recordsToSave];
```

Comment:
Letting target save to local store - so the local store doesn't have to be smartstore anymore.